src/multi-iterator.vala

```diff
@@ -0,0 +1,39 @@
+using GLib;
+
+public class Vast.MultiIterator : Object
+{
+    private Array[] _arrays;
+
+    public Array* arrays {
+        get { return _arrays; }
+        construct { /*_arrays = value;*/ }
+    }
+
+    [CCode (array_length = false)]
+    private ssize_t*[] cursor;
+
+    [CCode (array_length = false)]
+    private size_t[] offset;
+
+    public MultiIterator (Array[] arrays)
+    {
+        base (arrays: arrays);
+    }
+
+    [CCode (array_length = false)]
+    public (unowned void*)[]
+    get ()
+    {
+        var ret = new (unowned void*)[_arrays.length];
```

Comment:
We might want to avoid allocating here since it's always the same size. I guess it can be the caller responsibility to make a copy (a.k.a. unowned).